cluster/src/main/scala/io/snappydata/gemxd/SparkSQLExecuteImpl.scala

```diff
@@ -237,7 +246,21 @@ class SparkSQLExecuteImpl(val sql: String,
       case t: DecimalType => (StoredFormatIds.SQL_DECIMAL_ID, t.precision, t.scale)
       case FloatType => (StoredFormatIds.SQL_REAL_ID, -1, -1)
       case DoubleType => (StoredFormatIds.SQL_DOUBLE_ID, -1, -1)
-      case StringType => (StoredFormatIds.SQL_CLOB_ID, -1, -1)
+      case s: StringType =>
+        if (snc.sessionState.conf.stringAsClob || stringAsClob) {
+            if (f.metadata.contains(Constant.CHAR_SIZE_NAME) &&
```

Comment:
Please change the name 'CHAR_SIZE_NAME' to something more appropriate which conveys the intention.

Like: CHARTYPE_SIZE_PROP